Documentation/pre-built-detection.md

```diff
@@ -0,0 +1,147 @@
+# Pre-built detection guide
+
+This is a detailed guide on how to approach enabling pre-built detection for a source-buildable repository.
+It is primarily intended for repository maintainers in the `dotnet` organization.
+
+## Pre-requisites
+
+This guide places several assumptions on the repository and the maintainer in charge of enabling pre-built detection:
+
+  - the repository in question is source-buildable;
+  - the person performing the task is aware of basic source-build concepts;
+
+## Table of content
+  - [Pre-builts](#pre-builts)
+  - [Enabling mandatory pre-built detection](#enabling-mandatory-pre-built-detection)
+  - [Elimitating pre-builts](#eliminating-pre-builts)
+  - [FAQ](#faq)
+  - [Contacts](#contacts)
+
+# Pre-builts
+
+_Source-build_ is a process of building a given product on a single machine with no internet access.
+As such, all of the upstream packages must be part of the current source-build, since their _pre-built_ versions cannot be retrieved from, for example, `nuget.org` or some internal feed.
+
+By definition, _pre-builts_ are dependencies that a repo has on binary files that are not build from source, where _build from source_ points to any package produced during the _current source-build_.
```

Comment:
There is an exception here in the realm of tools such as cmake.  Toolset dependencies come from the distro in which .NET is being built on (which themselves were source-built).

Reply:
Good point, thanks. Rephrased and fixed